test/test_scheduler.py

```diff
@@ -2101,6 +2099,57 @@ async def test_offer_rescinded(self, fix: "TestScheduler.Fixture") -> None:
         assert await fix.driver_calls() == []
         launch.cancel()
 
+    async def test_unresolved_offer_rescinded(self, fix: "TestScheduler.Fixture", mocker) -> None:
+        """Test offerRescinded for an offer that is not yet resolved."""
+        mocker.patch.object(
+            asyncio.get_running_loop(), "getaddrinfo", side_effect=getaddrinfo_never
+        )
+        launch, kill = await fix.transition_node0(TaskState.STARTING)
+        offers = fix.make_offers()
+        fix.sched.resourceOffers(fix.driver, offers)
+        await exhaust_callbacks()
+        assert len(fix.sched._resolving_offers) == 2
+        assert fix.nodes[0].state == TaskState.STARTING
+        fix.sched.offerRescinded(fix.driver, offers[0].id)
+        fix.sched.offerRescinded(fix.driver, offers[1].id)
+        await exhaust_callbacks()
+        assert fix.sched._resolving_offers == {}
+        assert await fix.driver_calls() == []
+        launch.cancel()
+
+    async def test_offer_resolve_fail(self, fix: "TestScheduler.Fixture", mocker) -> None:
+        """Test receiving an offer that cannot be resolved with getaddrinfo."""
+        mocker.patch.object(asyncio.get_running_loop(), "getaddrinfo", side_effect=getaddrinfo_fail)
+        launch, kill = await fix.transition_node0(TaskState.STARTING)
+        offers = fix.make_offers()
+        fix.sched.resourceOffers(fix.driver, offers)
+        await asyncio.sleep(6)  # Give it a chance to try multiple times
+        assert fix.sched._resolving_offers == {}  # Must clean up the tasks
+        assert fix.nodes[0].state == TaskState.STARTING  # Must not start
+        assert await fix.driver_calls() == [
+            mock.call.declineOffer([offers[0].id]),
+            mock.call.declineOffer([offers[1].id]),
+        ]
+        launch.cancel()
+
+    async def test_decline_unresolved_offers(self, fix: "TestScheduler.Fixture", mocker) -> None:
+        """Test that when an unresolved offer is no longer needed, it is declined and removed."""
```

Comment:
If I'm really being nitpicky: There is some minor ambiguity in this docstring.
* Test name outlines `unresolved_offers` plural, which is what is actually tested.
* I guess the docstring/description refers to the general process of what happens to an unresolved offer.

I mainly mention this because I got the impression that one offer would resolve and the other would not.
* Might be the wrong impression, so please do correct me accordingly.

Reply:
Changed the docstring to plural.